src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -462,6 +532,44 @@ private static bool IsValueTypeErrorOrBlank<T>(FormulaValue val)
             return val is T || val is BlankValue || val is ErrorValue;
         }
 
+        private static FormulaValue DistinctValueType<TPFxPrimitive, TDotNetPrimitive>(List<(DValue<RecordValue> row, FormulaValue sortValue)> pairs, IRContext irContext)
+            where TPFxPrimitive : PrimitiveValue<TDotNetPrimitive>
+            where TDotNetPrimitive : IComparable<TDotNetPrimitive>
+        {
+            var values = new Dictionary<TDotNetPrimitive, DValue<RecordValue>>();
+            var name = ((TableType)irContext.ResultType).SingleColumnFieldName;
+            var type = ((TableType)irContext.ResultType).SingleColumnFieldType;
+
+            // Have to handle null
+            var addBlankValue = false;
+
+            foreach (var (row, sortValue) in pairs)
+            {
+                if (sortValue?.ToObject() == null)
```

Comment:
>sortValue?.ToObject() == null [](http://example.com/codeflow?start=20&length=29)

this can be rewritten as, `if(sortValue is BlankValue)`

Reply:
Updated.